scripts/van_allen_tools/set_up_vanallen_workspaces.py

```diff
@@ -0,0 +1,189 @@
+    ws_exists, ws_exists_response = check_workspace_exists(workspace_name, project)
+
+    if ws_exists is None:
+        return False, ws_exists_response
+
+    if not ws_exists:  # workspace doesn't exist (404), create workspace
+        # create request JSON
+        create_ws_json = make_create_workspace_request(workspace_name, project)  # json for API request
+
+        # request URL for createWorkspace (rawls) - bucketLocation not supported in orchestration
+        uri = f"https://rawls.dsde-prod.broadinstitute.org/api/workspaces"
+
+        # Get access token and and add to headers for requests.
+        # -H  "accept: application/json" -H  "Authorization: Bearer [token] -H  "Content-Type: application/json"
+        headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json", "Content-Type": "application/json"}
+
+        # capture response from API and parse out status code
+        response = requests.post(uri, headers=headers, data=json.dumps(create_ws_json))
+        status_code = response.status_code
+
+        if status_code != 201:  # ws creation fail
+            print(f"WARNING: Failed to create workspace with name: {workspace_name}. Check output file for error details.")
+            return False, response.text
+        # workspace creation success
+        print(f"Successfully created workspace with name: {workspace_name}.")
+        return True, None
+
+    # workspace already exists
+    print(f"Workspace already exists with name: {project}/{workspace_name}.")
+    print(f"Existing workspace details: {json.dumps(json.loads(ws_exists_response), indent=2)}")
+    # make user decide if they want to update/overwrite existing workspace
+    while True:  # try until user inputs valid response
+        update_existing_ws = input("Would you like to continue modifying the existing workspace? (Y/N)" + "\n")
+        if update_existing_ws.upper() in ["Y", "N"]:
+            break
+        else:
+            print("Not a valid option. Choose: Y/N")
+    if update_existing_ws.upper() == "N":       # don't overwrite existing workspace
+        deny_overwrite_message = f"{project}/{workspace_name} already exists. User selected not to overwrite. Try again with unique workspace name."
+        return None, deny_overwrite_message
+
+    accept_overwrite_message = f"{project}/{workspace_name} already exists. User selected to overwrite."
+    return True, accept_overwrite_message    # overwrite existing workspace - 200 status code for "Y"
+
+
+def make_create_workspace_request(workspace_name, project=NAMESPACE):
+    """Make the json request to pass into create_workspace()."""
+
+    # initialize empty dictionary
+    create_ws_request = {}
+
+    create_ws_request["namespace"] = project
+    create_ws_request["name"] = workspace_name
+    create_ws_request["attributes"] = {}
+    create_ws_request["noWorkspaceOwner"] = False
+    create_ws_request["bucketLocation"] = "us-central1"
```

Comment:
this is always for sure going to be `us-central1` ? any benefit in having this be the default but allowing the user to input a different location if desired?

Reply:
I thought about making it user modifiable but I think the lab wanted to be super careful and have all their resources to be in that region so as not to incur accidental egress - so i thought a hard coded region might be best! i could see how somewhere done the line at some point *maybe* they might want to change regions? what do you think? just make it a variable and set default?

Reply:
ah hm that makes sense. maybe then make it super obvious by making a top-level global variable `TARGET_BUCKET_LOCATION = "us-central1"` and then here `create_ws_request["bucketLocation"] = TARGET_BUCKET_LOCATION`

Reply:
sweet - same thought. just updated